tests/unit/presenters/admin/indexation-list-item-presenter-test.php

```diff
@@ -32,35 +84,66 @@ public function test_construct() {
 
+		// Assert.
 		$expected  = '<li><strong>SEO Data</strong>';
 		$expected .= '<p><a href="" target="_blank">Yoast SEO creates and maintains an index of all of your site\'s SEO data in order to speed up your site</a>.';
 		$expected .= ' To build your index, Yoast SEO needs to process all of your content.</p>';
 		$expected .= '<span class="wpseo-checkmark-ok-icon"></span>Great, your site has been optimized!';
 		$expected .= '</li>';
 
-		$this->assertSame( $expected, $instance->present() );
+		$this->assertSame( $expected, $result );
 	}
 
 	/**
 	 * Tests the case when there is something to index.
 	 *
 	 * @covers ::present
 	 */
-	public function test_present_with_something_to_index() {
-		$instance = new Indexation_List_Item_Presenter( 30 );
+	public function test_present_with_something_to_index_non_production() {
+		// Arrange.
+		$this->default_arrange( 30 );
+		$this->indexable_helper->shouldReceive( 'should_index_indexables' )->andReturn( false );
 
-		Monkey\Functions\expect( 'add_query_arg' )->andReturn( '' );
+		// Act.
+		$result = $this->instance->present();
+
+		// Assert.
+		$expected  = '<li><strong>SEO Data</strong>';
+		$expected .= '<p><a href="" target="_blank">Yoast SEO creates and maintains an index of all of your site\'s SEO data in order to speed up your site</a>. ';
+		$expected .= 'To build your index, Yoast SEO needs to process all of your content.</p>';
+		$expected .= '<span id="yoast-indexation"><button type="button" class="button yoast-open-indexation" data-title="Speeding up your site" data-settings="yoastIndexationData" disabled>Start processing and speed up your site now</button></span>';
+		$expected .= '<p>This button to index your website is disabled for non-production environments.</p>';
+		$expected .= '</li>';
+
+		$this->assertSame( $expected, $result );
+	}
+
+	/**
+	 * Tests the case when there is something to index.
+	 *
+	 * @covers ::present
+	 */
+	public function test_present_with_something_to_index_production() {
+		// Arrange.
+		$this->default_arrange( 30 );
+		$this->indexable_helper->shouldReceive( 'should_index_indexables' )->andReturn( true );
+
+		// Act.
+		$result = $this->instance->present();
 
+		// Assert.
 		$expected  = '<li><strong>SEO Data</strong>';
 		$expected .= '<p><a href="" target="_blank">Yoast SEO creates and maintains an index of all of your site\'s SEO data in order to speed up your site</a>. ';
 		$expected .= 'To build your index, Yoast SEO needs to process all of your content.</p>';
-		$expected .= '<span id="yoast-indexation"><button type="button" class="button yoast-open-indexation" data-title="Speeding up your site" data-settings="yoastIndexationData">Start processing and speed up your site now</button></span>';
+		$expected .= '<span id="yoast-indexation"><button type="button" class="button yoast-open-indexation" data-title="Speeding up your site" data-settings="yoastIndexationData" >Start processing and speed up your site now</button></span>';
```

Comment:
Hmm, this feels not okay. We should fix the place where the space was added instead of fixing the test :)

Reply:
yes, this is bad. let's move the discussion to the point where [the problem](https://github.com/Yoast/wordpress-seo/pull/16049#discussion_r490095033) is introduced.